packages/src/event.test.tsx

```diff
@@ -74,4 +74,70 @@ describe('overlay object', () => {
     expect(screen.queryByText(testContent3)).toBeInTheDocument();
     expect(screen.queryByText(testContent4)).toBeInTheDocument();
   });
+
+  it('The value passed as an argument to close is passed to resolve. overlay.openAsync', async () => {
+    const wrapper = ({ children }: PropsWithChildren) => <OverlayProvider>{children}</OverlayProvider>;
+    const testContent = 'context-modal-test-content';
+    const dialogContent = 'context-modal-dialog-content';
+    const mockFn = vi.fn();
+    const Component = () => {
+      const handleClick = async () => {
+        const result = await overlay.openAsync<boolean>(({ close }) => (
+          <button onClick={() => close(true)}>{dialogContent}</button>
+        ));
+
+        if (result) {
+          mockFn(result);
+        }
+      };
+
+      return <button onClick={handleClick}>{testContent}</button>;
+    };
+
+    const renderComponent = render(<Component />, { wrapper });
+    const testContentElement = await renderComponent.findByText(testContent);
+
+    act(() => {
+      testContentElement.click();
+    });
+
+    const dialogContentElement = await renderComponent.findByText(dialogContent);
+
+    act(() => {
+      dialogContentElement.click();
+    });
+    waitFor(() => {
+      expect(mockFn).toHaveBeenCalledWith(true);
+    });
+  });
+
+  it('should be able to turn off overlay through close overlay.openAsync', async () => {
+    const wrapper = ({ children }: PropsWithChildren) => <OverlayProvider>{children}</OverlayProvider>;
+    const testContent = 'context-modal-test-content';
+    const dialogContent = 'context-modal-dialog-content';
+
+    const Component = () => {
+      const handleClick = async () => {
+        overlay.openAsync<boolean>(({ close }) => <button onClick={() => close(true)}>{dialogContent}</button>);
+      };
+      return <button onClick={handleClick}>{testContent}</button>;
+    };
+
+    const renderComponent = render(<Component />, { wrapper });
+    const testContentElement = await renderComponent.findByText(testContent);
+
+    act(() => {
+      testContentElement.click();
+    });
+
+    const dialogContentElement = await renderComponent.findByText(dialogContent);
+
+    act(() => {
+      dialogContentElement.click();
+    });
+
+    waitFor(() => {
+      expect(dialogContentElement).not.toBeInTheDocument();
+    });
```

Comment:
I thought it was odd that `await` wasn't attached here.
I'll share my thoughts as I test it out. If I'm wrong, feel free to tell me. 🙇‍♂️

## Test fails when `await` is appended

I only added `await` in the code you provided, so the test fails.

```tsx
await waitFor(() => {
  expect(dialogContentElement).not.toBeInTheDocument();
});
```

![스크린샷 2024-07-15 오전 11 24 58](https://github.com/user-attachments/assets/ee771ceb-dc17-4a71-b6cb-db1289541ea5)

I think the reason the test passed without the `await` is because the test inside the `waitFor` works asynchronously, but the test function doesn't wait for it.

So if I don't add `await`, I think I can write a test like this and it will pass.

```tsx
waitFor(() => expect(dialogContentElement).toBeInTheDocument());
```

![스크린샷 2024-07-15 오전 11 14 00](https://github.com/user-attachments/assets/fa1b7b4f-8cdb-4555-97d5-d7e60af006ff)

## Why does it fail?

The next question is where the test fails when you add `await`.
`close` is not removed from the `DOM`, so `not.toBeInTheDocument()` seems appropriate with `unmount`.

I can't think of a way to test the `close` behavior right now. sorry. 😂
I'll have to think about that.

Reply:
You are right. It was my mistake to omit the "await" and the test would return a false positive.

Thank you for catching my mistake

The reason it fails is because "close" does not actually unmount the element, but rather changes the value of isOpen.

Therefore, I modified the code as follows:

```tsx
      const handleClick = async () => {
        overlay.openAsync<boolean>(({ isOpen, close }) =>
          // focus on isOpen
          isOpen ? <button onClick={() => close(true)}>{dialogContent}</button> : null
        );
      };
...
    await waitFor(() => {
      expect(dialogContentElement).not.toBeInTheDocument();
    });
```
This code allows the button to be unmounted from the screen based on the value of isOpen.

A commit that resolves the issue has been pushed to this PR.

Thank you for checking carefully.  With your help I was able to find this problem